flexget/plugins/api/seen.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import unicode_literals, division, absolute_import
+
+from math import ceil
+from operator import itemgetter
+from urllib import unquote
+
+from flask_restplus import inputs
+
+from flexget.api import api, APIResource, jsonify, request
+from flexget.plugins.filter import seen
+from flexget.utils.imdb import is_imdb_url, extract_id
+
+PLUGIN_TASK_NAME = 'seen_plugin_API'
+
+
+def return_imdb_id(value):
+    if is_imdb_url(value):
+        imdb_id = extract_id(value)
+        if imdb_id:
+            value = imdb_id
+    return value
+
+
+seen_api = api.namespace('seen', description='Managed Flexget seen entries and fields')
+
+seen_field_object = {
+    'type': 'object',
+    'properties': {
+        'id': {'type': 'integer'},
+        'field': {'type': 'string'},
+        'value': {'type': 'string'},
+        'added': {'type': 'string'},
+        'seen_entry_id': {'type': 'integer'}
+    }
+}
+
+seen_object = {
+    'type': 'object',
+    'properties': {
+        'id': {'type': 'integer'},
+        'title': {'type': 'string'},
+        'reason': {'type': 'string'},
+        'task': {'type': 'string'},
+        'added': {'type': 'string'},
+        'local': {'type': 'string'},
+        'fields': {'type': 'array', 'items': seen_field_object}
+    }
+}
+seen_object_schema = api.schema('seen_object_schema', seen_object)
+
+seen_object_input_schema = {
+    'type': 'object',
+    'properties': {
+        'title': {'type': 'string'},
+        'reason': {'type': 'string'},
+        'task': {'type': 'string'},
+        'local': {'type': 'boolean', 'default': False},
+        'fields': {'type': 'object'}
```

Comment:
I guess I could but that's not very readable. Is there a problem with how it's presented now?